### PR TITLE
[9.x] Align CSRF cookie duration to session cookie duration

### PR DESCRIPTION
**Pull request by lorenzolosa**
**TL;DR**: currently, the duration of session cookies and CSRF cookie is misaligned, resulting in the possibility that one expires while the other is still valid. This pull requests fixes this by slightly changing the way in which the CSRF cookie is set.

The duration of the session cookie is defined by two configuration variables: `session.lifetime` and `session.expire_on_close`. If `session.expire_on_close` is `true`, then the cookie expires on close; if it is `false` (default), then the cookie has a duration of `session.lifetime` minutes (this happens in the `getCookieExpirationDate` method of `Illuminate\Session\Middleware\StartSession`).

The CSRF cookie (`XSRF-TOKEN`, set by the `VerifyCsrfToken` middleware) uses the same duration of the session cookies (`session.lifetime`), but the middleware does not consider the `session.expire_on_close` variable 
(this happens in the `addCookieToResponse` method of `Illuminate\Foundation\Http\Middleware\VerifyCsrfToken`).

This is an unexpected behaviour, and looks like a bug. As a result, if `session.expire_on_close` is `true` it is possible that the session cookie expires while the CSRF cookie is still valid, or that the session cookie is still valid while the CSRF cookie expires. None of these seems a desirable behaviour.

This pull request aligns the behaviour of the CSRF cookie to the one of the session cookie, by taking into account the value of the `session.expire_on_close` variable.

I originally wrote a pull request for the 8.x branch (#37267), where it was suggested that it was not a change for the stable branch, and therefore I'm submitting it for the 9.x release.

### CODE REVIEW

**Comment by taylorotwell** [comment]
There is not really a downside to the session cookie expiring before the CSRF cookie. Is there some practical problem you wish to fix?

**Comment by lorenzolosa** [comment]
(Note that both can happen - the session cookie expiring before the CSRF cookie, or the CSRF cookie expiring before the session cookie)

The current behaviour feels strange to me, because there are two similar things that works almost in the same way, but not exactly in the same way (they share one config parameters but not the other). This is often risky. However, I don't think this actually breaks anything: if it is indeed the desired behaviour, well, good!

The reason why I looked into this is because I've been on a project where, from the point of view of the Cookie Policy, session cookies are preferred. I then switched the `session` to be a session cookie, and I was surprised to see that the CSRF cookie was not affected (and therefore I extended VerifyCsrfToken to change that).
